pytd/writer.py

```diff
@@ -49,10 +63,23 @@ def _cast_dtypes(dataframe, inplace=True, keep_list=False):
             t = "Int64" if df[column].isnull().any() else "int64"
         elif kind == "f":
             t = float
-        elif kind == "O" and keep_list:
-            if df[column].apply(lambda x: isinstance(x, (list, np.ndarray))).all():
+        elif kind == "O":
+            if df[column].apply(_check_column_type, args=[(list, np.ndarray)]).all():
+                t = object
+                if keep_list:
+                    df[column] = df[column].apply(_to_list)
+                else:
+                    df[column] = df[column].apply(
+                        _convert_nullable_str, args=[(list, np.ndarray)]
+                    )
+            elif df[column].apply(_check_column_type, args=[bool]).all():
```

Comment:
Then we don't need these lines, right?
https://github.com/treasure-data/pytd/blob/a109ccd1141dcef6fb7fd5534165e5bd7cab15c2/pytd/writer.py#L58-L62